packages/truffle-contract/test/separation.js

```diff
@@ -1,54 +1,48 @@
 var assert = require("chai").assert;
-var util = require('./util');
+var util = require("./util");
 
 describe("instance-instance / instance-constructor separation", function() {
   var Example;
   var exampleA;
   var exampleB;
-  var accounts;
-  var web3;
-  var providerOptions = {vmErrorsOnRPCResponse: false};
+  var providerOptions = { vmErrorsOnRPCResponse: false };
 
   before(function() {
     this.timeout(10000);
 
     Example = util.createExample();
 
-    return util
-      .setUpProvider(Example, providerOptions)
-      .then(result => {
-        web3 = result.web3;
-        accounts = result.accounts;
-      });
+    return util.setUpProvider(Example, providerOptions);
   });
 
   beforeEach(async () => {
     exampleA = await Example.new(1);
     exampleB = await Example.new(2);
   });
 
-  it('instances should not hear each others events', function(done){
+  it("instances should not hear each others events", function(done) {
     const events = [];
     const event = exampleA.ContractAddressEvent();
 
-    event.on('data', evt => {
+    event.on("data", evt => {
       events.push(evt);
-      if (evt.args._contract === exampleA.address){
+      if (evt.args._contract === exampleA.address) {
         assert(events.length === 1);
         done();
       }
     });
 
     exampleB
       .triggerContractAddressEvent()
-      .then(exampleA.triggerContractAddressEvent);
+      .then(() => exampleA.triggerContractAddressEvent())
+      .catch(assert.fail);
```

Comment:
This test caused an unhandled promise rejection when it was failing. Now it doesn't.